nmostesting/GenericTest.py

```diff
@@ -383,6 +397,97 @@ def validate_schema(self, payload, schema):
+        cert_ext.append(crypto.X509Extension(b'keyUsage', True, b'digitalSignature, keyEncipherment'))
+        cert_ext.append(crypto.X509Extension(b'subjectAltName', False, ','.join(san).encode()))
+        cert.add_extensions(cert_ext)
+
+        # sign cert with Intermediate CA key
+        cert.sign(ca_pkey, 'sha256')
+
+        # write chain certificate file
+        if cert_file is not None:
+            with open(cert_file, "wt") as f:
+                f.write(crypto.dump_certificate(crypto.FILETYPE_PEM, cert).decode("utf-8"))
+                f.write(cacert)
+        # write private key file
+        if key_file is not None:
+            with open(key_file, "wb") as f:
+                pem = k.to_cryptography_key().private_bytes(
+                    encoding=serialization.Encoding.PEM,
+                    format=serialization.PrivateFormat.TraditionalOpenSSL,
+                    encryption_algorithm=serialization.NoEncryption()
+                )
+                f.write(pem)
+
+    def start_secondary_authorization_server(self, auth):
+        ctx = None
+        # placeholder for the certificate
+        cert_file = "test_data/BCP00301/ca/mock_auth_cert.pem"
+        # placeholder for the private key
+        key_file = "test_data/BCP00301/ca/mock_auth_private_key.pem"
+        # generate RSA key and certificate for the mock secondary Authorization server
+        self.make_key_cert_files(cert_file, key_file)
+        if CONFIG.ENABLE_HTTPS:
+            # ssl.create_default_context() provides options that broadly correspond to the requirements of BCP-003-01
+            ctx = ssl.create_default_context()
+            ctx.load_cert_chain(cert_file, key_file)
+            # additionally disable TLS v1.0 and v1.1
+            ctx.options &= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1
+            # BCP-003-01 however doesn't require client certificates, so disable those
+            ctx.check_hostname = False
+            ctx.verify_mode = ssl.CERT_NONE
+
+        # mock secondary Authorization server
+        auth_app = Flask(__name__)
+        auth_app.debug = False
+        auth_app.config["AUTH_INSTANCE"] = 1
+        auth.private_keys = [key_file]
+        port = auth.port
+        auth_app.register_blueprint(AUTH_API)
+
+        self.secondary_authorization_server = make_server("0.0.0.0", port, auth_app, threaded=True, ssl_context=ctx)
+        self.secondary_authorization_server_thread = threading.Thread(
+            target=self.secondary_authorization_server.serve_forever)
+        self.secondary_authorization_server_thread.daemon = True
+        self.secondary_authorization_server_thread.start()
+
+    def stop_secondary_authorization_server(self):
+        if self.secondary_authorization_server_thread:
+            self.secondary_authorization_server.shutdown()
+            self.secondary_authorization_server_thread.join()
+            self.secondary_authorization_server_thread = None
```

Comment:
Better to move this into it's own object containing the Auth server state and moved out of `GenericTest.py` into `Auth.py`

Reply:
Done, new `AuthServer` class in `Auth.py`